src/content/profile-tree.js

```diff
@@ -109,16 +120,21 @@ class ProfileTree {
     if (node === undefined) {
       const funcIndex = this._funcStackTable.func[funcStackIndex];
       const funcName = this._stringTable.getString(this._funcTable.name[funcIndex]);
+      const resourceIndex = this._funcTable.resource[funcIndex];
+      const resourceType = this._resourceTable.type[resourceIndex];
       const isJS = this._funcTable.isJS[funcIndex];
+      const libName = this._getOriginAnnotation(funcIndex);
 
       node = {
         totalTime: `${this._funcStackTimes.totalTime[funcStackIndex].toFixed(1)}ms`,
         totalTimePercent: `${(100 * this._funcStackTimes.totalTime[funcStackIndex] / this._rootTotalTime).toFixed(1)}%`,
         selfTime: `${this._funcStackTimes.selfTime[funcStackIndex].toFixed(1)}ms`,
         name: funcName,
-        lib: this._getOriginAnnotation(funcIndex),
+        lib: libName,
         // Dim platform pseudo-stacks.
         dim: !isJS && this._jsOnly,
+        // http(s) resources should have "webhost" type but currently all URLs are of type "url". Let's be future proof.
+        icon: [resourceTypes.url, resourceTypes.webhost].includes(resourceType) ? extractFaviconFromLibname(libName) : null,
```

Comment:
yay, I saw something changed when I had to sresolve conflicts but hadn't realized this actually brought this :)